viewing_party/party.py

```diff
@@ -1,23 +1,179 @@
+        for key, value in movie.items():
+            if key == "genre":
+                if value not in genre_map:
+                    genre_map[value] = 1
+                else:
+                    genre_map[value] += 1
+    for genre, num in genre_map.items():
+        if most_genre == None:
+            most_genre = genre
+            highest_num = num
+        elif num > highest_num:
+            most_genre = genre
+            highest_num = num
+        else:
+            continue
 
 
+    return most_genre
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_movies = []
+    friend_movies = get_friend_movie_lists(user_data)
+    user_movies = user_data["watched"]
+    for movie_dict in user_movies:
+        if movie_dict not in friend_movies:
+            unique_movies.append(movie_dict)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    unique_movies = []
+    friend_movies = get_friend_movie_lists(user_data)
+    user_movies = user_data["watched"]
+    for movie_dict in friend_movies:
+        if movie_dict not in user_movies:
+            unique_movies.append(movie_dict)
+
+    return unique_movies
+
+def get_friend_movie_lists(user_data):
+    friend_movies = []
+    for friend in user_data["friends"]:
+        for movie_dict in friend["watched"]:
+            if movie_dict not in friend_movies:
+                friend_movies.append(movie_dict)
+    return friend_movies
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    all_available_recs = get_friends_unique_watched(user_data)
+    user_subscriptions = get_user_subscription(user_data)
+    recs_by_subscription = []
+    for subscription in user_subscriptions:
+        for movie_dict in all_available_recs:
+            if subscription in movie_dict["host"]:
+                recs_by_subscription.append(movie_dict)
+    return recs_by_subscription
+
+def get_user_subscription(user_data):
+    subs = []
+    if "subscriptions" in user_data:
+        for subscription in user_data["subscriptions"]:
+            subs.append(subscription)
+    return subs
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    fav_genre_recs = []
+    fav_genre = most_frequent_genre(user_data)
+
+    if fav_genre is None:
+        return fav_genre_recs
+
+    available_recs_by_subs = get_available_recs(user_data)
+
+    for movie_dict in available_recs_by_subs:
+        if movie_dict["genre"] == fav_genre:
+            fav_genre_recs.append(movie_dict)
+        else:
+            continue
+
+    return fav_genre_recs
+
+def most_frequent_genre(user_data):

```

Comment:
This takes a slightly different approach but does it return anything different from `get_most_watched_genre`? I substituted the function call for the one from Wave 2 and the tests still pass. We can reuse that function and no have to write another that does the same thing!